Library/Homebrew/extend/fileutils.rb

```diff
@@ -10,6 +11,19 @@ def mktemp(prefix = name)
     prev = pwd
     tmp  = Dir.mktmpdir(prefix, HOMEBREW_TEMP)
 
+    # Make sure files inside the temporary directory have the same group as the
+    # brew instance.
+    #
+    # Reference from `man 2 open`
+    # > When a new file is created, it is given the group of the directory which
+    # contains it.
+    if (File.grpowned? HOMEBREW_BREW_FILE)
+      chown(nil, File.stat(HOMEBREW_BREW_FILE).gid, tmp)
+    else
+       #chown(nil, Etc.getpwnam(Etc.getlogin).gid, tmp)
+       chown(nil, Process.gid, tmp)
```

Comment:
> unless the idea is to support some kind of "advanced" (read: confused) configuration where bin/brew and Library are all admin and Cellar is all staff or something weird like that.

Ideally we'd not support this but I'm 100% sure there are users for whom that configuration is working fine just now and I'd rather keep it working for them if we can.